pkg/upgrade/upgrade.go

```diff
@@ -368,41 +370,49 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 	return nil
 }
 
-func CleanupExistingResource(cli client.Client, platform deploy.Platform) error {
+// TODO: remove function once we have a generic solution across all components.
+func CleanupExistingResource(cli client.Client, platform deploy.Platform, dscApplicationsNamespace, dscMonitoringNamespace string) error {
 	var multiErr *multierror.Error
+	ctx := context.TODO()
```

Comment:
If you wrap this call in `manager.RunnableFunc`  you will have a ctx from manager, which you can pass by adding first param here.